sdk/javascript/src/RuleBasedTransactionFactory.js

```diff
@@ -43,7 +43,11 @@ const typeConverterFactory = (module, customTypeConverter, value) => {
 	if (value instanceof ByteArray) {
 		/** @type object */
 		const ByteArrayClass = value.constructor;
-		return new module[ByteArrayClass.NAME](value.bytes);
+
+		if (ByteArrayClass.NAME in module)
+			return new module[ByteArrayClass.NAME](value.bytes);
+
+		return new ByteArrayClass(value.bytes);
```

Comment:
Is there a case where the ``NAME`` is not set in the class?

Reply:
yes - it is only set in the public types, but not the generated model types